rfcs/text/0000-alternative-planning.md

```diff
@@ -0,0 +1,315 @@
+
+1) We find that the implementation doesn't deliver what the user needs. We thought we understood the problem but we actually didn't 
+because we didn't test that understanding against reality. Now we have to cycle around for
+another quarter (or even longer!) whilst we amend the design.
+2) The design no longer meshes with the rest of the DCP. Other components have moved on and now this design needs significant
+rework. If this is an RFC then that's another quarter or more gone.
+3) The people who wrote the design have moved on (or simply forgotten their original ideas). RFCs are written documents, 
+not code, so they are not guaranteed to capture the essence of the problem or the solution. Reviewers can't catch 
+these because the DCP is too complex for anyone to grasp all the details. We no longer know the full thinking behind the
+solution or critical fine implementation detail.
+4) The science has changed so what we planned is no longer relevant. On the one hand this may be a good thing - we didn't
+spend implementation time on it. On the other it may be bad - we spent time planning something when we could have been
+doing other work of surer value instead.
+5) The design was still relevant. It needed no major revisions. On implementation we find it accurately addresses the user's need
+with little or no need for corrective work to fulfill the user story. Everybody is happy.
+
+In my experience, the last scenario is the rarest case. To me, this is a big reason for why agile came about in the
+first place - from the need to be adaptable to customer requirements and to fail fast so we can find what does work. A 
+process where major work takes a half a year or longer to get from planning to execution is not an agile process. 
+  
+Nowadays it's pretty much orthodoxy that nobody does waterfall unless they must (a contentious statement perhaps
+though I've seen that sentiment in RFC reviews). So why is it being proposed by the Planning and Roadmaps RFC? From my 
+perspective it's because of these challenges:
+
+* **High complexity** - the DCP feels highly complex. This makes for a very difficult management problem. Everything is
+connected.
+* **Siloed teams** - A team works on and owns its own components. Getting other teams to act means negotiation. It's
+ much easier to do a local solution that is sub-optimal for the DCP as a whole than a global one where some teams may 
+ need to change their chosen design.
+* **Constrained communication** - plenary voice communication is tightly constrained to an hour per day. This constrains
+communication both between DCP groups with the same role (e.g. ingest, analysis, azul dev groups) and up and down
+the objective to implementation work stack (e.g. between developers and product owners, between product owners and 
+project leads).
+* **No overall control** - no organization is in overall control.
```

Comment:
What role does the Oversight Committee play in this challenge?

Reply:
Ensure that the aims make sense.